cmd/template/cluster/provider/aws.go

```diff
@@ -76,3 +171,51 @@ func WriteAWSTemplate(out io.Writer, config ClusterCRsConfig) error {
 
 	return nil
 }
+
+func getCAPAClusterTemplate(config ClusterCRsConfig) (client.Template, error) {
+	var err error
+
+	c, err := client.New("")
+	if err != nil {
+		return nil, err
+	}
+
+	templateOptions := client.GetClusterTemplateOptions{
+		ClusterName:       config.ClusterID,
+		TargetNamespace:   key.OrganizationNamespaceFromName(config.Owner),
+		KubernetesVersion: "v1.19.9",
+		ProviderRepositorySource: &client.ProviderRepositorySourceOptions{
+			InfrastructureProvider: "aws:v0.6.6",
+			Flavor:                 "machinepool",
+		},
+	}
+	os.Setenv("AWS_SUBNET", "")
+	os.Setenv("AWS_CONTROL_PLANE_MACHINE_TYPE", "")
+	os.Setenv("AWS_REGION", "")
+	os.Setenv("AWS_SSH_KEY_NAME", "")
+
+	if replicas := int64(len(config.ControlPlaneAZ)); replicas > 0 {
+		templateOptions.ControlPlaneMachineCount = &replicas
+	}
+
+	clusterTemplate, err := c.GetClusterTemplate(templateOptions)
+	if err != nil {
+		return nil, err
+	}
+	return clusterTemplate, nil
+}
+
+func newAWSMachineTemplateFromUnstructured(config ClusterCRsConfig, o unstructured.Unstructured) (*capav1alpha3.AWSMachineTemplate, error) {
+	var awsmachinetemplate capav1alpha3.AWSMachineTemplate
+	{
+		err := runtime.DefaultUnstructuredConverter.
+			FromUnstructured(o.Object, &awsmachinetemplate)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		if config.ControlPlaneAZ != nil {
+			awsmachinetemplate.Spec.Template.Spec.FailureDomain = &config.ControlPlaneAZ[0]
```

Comment:
This is unfortunate but it seems like we can only have **one** AZ for the control-plane, even if several masters are running there. :( Not sure what is the most elegant way to deal with it.

Reply:
Hmm maybe we could try something else. If I got it right from the docs, you can specify `networkSpec` in order to select your AZ's for control-plane, if you don't want to have random AZ's, https://cluster-api-aws.sigs.k8s.io/topics/multi-az-control-planes.html#overview. This is set in the AWSCluster.

With this approach we would basically force all master nodes to be in the same region, not sure if we want that as you said 🤔

Reply:
Oh nice, I did not see that! :) I'll try to use this.